improver/calibration/reliability_calibration.py

```diff
@@ -1324,3 +1326,126 @@ def process(self, forecast: Cube, reliability_table: Union[Cube, CubeList]) -> C
             warnings.warn(msg)
 
         return calibrated_forecast
+
+    def _apply_point_by_point_calibration(
+        self, forecast: Cube, reliability_table: CubeList,
+    ) -> Cube:
+        """
+        Apply point by point reliability calibration by iteratively picking a spatial
+        coordinate within the forecast cube, extracting the forecast at that point
+        and the reliability table corresponding to that point, then passing the
+        extracted forecast and reliability table to _get_calibrated_forecast().
+
+        Args:
+            forecast:
+                The forecast to be calibrated.
+            reliability_table:
+                The reliability table to use for applying calibration.
+
+        Returns:
+            The forecast cube following calibration.
+        """
+
+        calibrated_cubes = iris.cube.CubeList()
+        y_name = forecast.coord(axis="y").name()
+        x_name = forecast.coord(axis="x").name()
+
+        # create list of dimensions other than time dimension
+        # and threshold dimension
+        dim_names = get_dim_coord_names(forecast)
+        dim_associated_coords = {}
+
+        # create lists with first entry a dimension name and second
+        # entry a list of auxiliary coordinates associated with that
+        # dimension. Collect these lists together into a list.
+        for dim_name in dim_names:
+            template_dim = [x for x in forecast.coord_dims(dim_name)]
+            associated_coords = [
+                c
+                for d in template_dim
+                for c in forecast.coords(dimensions=d, dim_coords=False)
+            ]
+            dim_associated_coords[dim_name] = associated_coords
+
+        # slice over the spatial dimension/s of the forecast cube
+        # and apply reliability calibration separately to each slice
+        # using a slice of the input reliability table at the same
+        # spatial point
+        for forecast_point in forecast.slices_over([y_name, x_name]):
+            y_point = forecast_point.coord(y_name).points[0]
+            x_point = forecast_point.coord(x_name).points[0]
+
+            # create reliability table containing only those cubes
+            # relating to the currently considered spatial point
+
+            reliability_table_point = reliability_table.extract(
+                iris.Constraint(coord_values={y_name: y_point, x_name: x_point})
+            )
```

Comment:
I think that there could be multiple reliability tables for each spatial point, for example one cube per threshold at each point, so that `.extract()` is required rather than `.extract_cube()`

Reply:
Ah yes. You're quite right. the _apply_reliabiltiy_calibration will do the final bit of extraction.